tools/btor2/btor2i/src/ir.rs

```diff
@@ -0,0 +1,473 @@
+// }
+
+// pub struct Btor2Instr {
+//     pub id: i64,
+//     pub linenum: u64,
+//     pub name: String,
+//     pub tag: Btor2Tag,
+//     pub sort: Btor2InstrSort,
+//     pub init: i64,
+//     pub next: u64,
+//     pub constant: String,
+//     pub symbol: String,
+//     pub nargs: u32,
+//     pub margs: u32,
+//     pub args: Vec<i64>,
+// }
+
+pub enum Btor2InstrContents {
+    Constant {
+        constant: Option<String>,
+        kind: ConstantType,
+    },
+    Literal {
+        kind: LiteralType,
+    },
+    UnOp {
+        arg1: i64,
+        kind: UnOpType,
+    },
+    BinOp {
+        arg1: i64,
+        arg2: i64,
+        kind: BinOpType,
+    },
+    Conditional {
+        arg1: i64,
+        arg2: i64,
+        arg3: i64,
+    },
+    Slice {
+        arg1: i64,
+        u: i64,
+        l: i64,
+    },
+    Input {
+        name: String,
+    },
+    Output {
+        name: String,
+        arg1: i64,
+    },
+    Sort,
+    Unknown, // TODO: this is very janky but will remove once we add interpreter support for state
+}
+
+pub struct Btor2Instr {
+    pub id: i64,
+    pub sort: SortType,
+    pub contents: Btor2InstrContents,
+}
+
+impl From<&Btor2Line<'_>> for Btor2Instr {
+    fn from(line: &Btor2Line) -> Btor2Instr {
+        let id = line.id();
+        let sort = SortType::from(line.sort().content());
+        // eprintln!("{:?}", line);
+        match line.tag() {
+            // core
+            btor2tools::Btor2Tag::Sort => Btor2Instr {
+                id,
+                sort,
+                contents: Btor2InstrContents::Sort,
+            }, // skip - sort information is handled by the parser
+            btor2tools::Btor2Tag::Const => convert_const_op(line),
+            btor2tools::Btor2Tag::Constd => convert_const_op(line),
+            btor2tools::Btor2Tag::Consth => convert_const_op(line),
+            btor2tools::Btor2Tag::Input => convert_input(line), // handled in parse_inputs
+            btor2tools::Btor2Tag::Output => convert_output(line), // handled in extract_output
+            btor2tools::Btor2Tag::One => convert_literal_op(line),
+            btor2tools::Btor2Tag::Ones => convert_literal_op(line),
+            btor2tools::Btor2Tag::Zero => convert_literal_op(line),
+
+            // indexed
+            btor2tools::Btor2Tag::Sext => convert_unary_op(line),
+            btor2tools::Btor2Tag::Uext => convert_unary_op(line),
+            btor2tools::Btor2Tag::Slice => convert_slice_op(line),
+
+            // unary
+            btor2tools::Btor2Tag::Not => convert_unary_op(line),
+            btor2tools::Btor2Tag::Inc => convert_unary_op(line),
+            btor2tools::Btor2Tag::Dec => convert_unary_op(line),
+            btor2tools::Btor2Tag::Neg => convert_unary_op(line),
+            btor2tools::Btor2Tag::Redand => convert_unary_op(line),
+            btor2tools::Btor2Tag::Redor => convert_unary_op(line),
+            btor2tools::Btor2Tag::Redxor => convert_unary_op(line),
```

Comment:
These arms can all be collapsed into a single pattern match, i.e. the or `|` of
the possible patterns. This is true of several other sections. It's not critical
but I think it is slightly cleaner